### PR TITLE
Submit my collection of scripts from my dotfiles.git

### PR DESCRIPTION
**Pull request by avar**
These are all irssi scripts that I maintain & use myself that I think
are useful for general use, none of this has anything hardcoded related
to me and when applicable has a configurable interface.

### CODE REVIEW

**Comment by dequis** [comment]
>bitlbee_hide_password.pl

You could just use /^msg (which sends a message without printing it). Or server password. Or sasl.

**Comment by avar** [comment]
@dequis Yeah I could do that, there's also "/OPER" which does a similar thing, but the reason I have that script is that just typing `identify` in the window is burned into my muscle memory, the script is also handy for anyone else used to doing the same.

File: scripts/bitlbee_hide_password.pl [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    # We only care about the &bitlbee control channel
+    my $target = $tdest->{target};
+    return unless $target eq '&bitlbee';
+
+    # We're matching against $stripped but replacing both because the
+    # $data thing is escaped and much harder to match against.
+    #
+    # We're just replacing nick mentions, so e.g. if you say "Hi I'm
+    # bob here but my username is bobby" it won't turn into "Hi I'm
+    # bobby here but my username is bobby".
```
Comment (ailin-nemui) [inline scripts/bitlbee_hide_password.pl]: I think you c&pd the comments but now they're in the wrong script and sort of confusing

File: scripts/munge_own_nickname_to_username.pl [hunk 2]
```diff
@@ -0,0 +1,74 @@
...
+    # $data thing is escaped and much harder to match against.
+    #
+    # We're just replacing nick mentions, so e.g. if you say "Hi I'm
+    # bob here but my username is bobby" it won't turn into "Hi I'm
+    # bobby here but my username is bobby".
+    #
+    # The illusion here isn't complete, e.g. if you do /NAMES your
+    # nick will show up and not your username, but I consider that a
+    # feature.
+    if ($stripped =~ /^<.?\Q$server_nick\E> /s) {
```
Comment (ailin-nemui) [inline scripts/munge_own_nickname_to_username.pl]: your script only works when the theme uses <nick> format, but it's difficult to do it better in currents' irssi

**Comment by avar** [comment]
@ailin-nemui Thanks a lot for the feedback. I fixed the confusing comment (and fixed the s/hawe/have/ spelling error there).

The second issue with the nick matching regex is trickier as you say, for now I've just added documentation in the comment alerting users to the  issue.

**Pull request merged**